app/src/main/java/org/mozilla/reference/browser/BrowserApplication.kt

```diff
@@ -34,6 +37,28 @@ open class BrowserApplication : Application() {
 
         components.core.engine.warmUp()
 
+        try {
+            WebExtensionSupport.initialize(
+                engine = components.core.engine,
+                store = components.core.store,
+                onNewTabOverride = { _, engineSession, url ->
+                    val session = Session(url)
+                    components.core.sessionManager.add(session, true, engineSession)
+                    session.id
+                },
+                onCloseTabOverride = { _, sessionId ->
+                    components.useCases.tabsUseCases.removeTab(sessionId)
+                },
+                onSelectTabOverride = { _, sessionId ->
+                    val selected = components.core.sessionManager.findSessionById(sessionId)
+                    selected?.let { components.useCases.tabsUseCases.selectTab(it) }
+                }
+            )
+        } catch (e: UnsupportedOperationException) {
+            // Web extension support is only available for engine gecko
```

Comment:
nit: We only have the latest gecko nightly in r-b so we don't need to handle this here?

Reply:
Ah, good point!